test/common/http/header_map_impl_speed_test.cc

```diff
@@ -295,5 +297,70 @@ static void headerMapImplRemovePrefix(benchmark::State& state) {
 }
 BENCHMARK(headerMapImplRemovePrefix)->Arg(0)->Arg(1)->Arg(5)->Arg(10)->Arg(50);
 
+class StaticLookupBenchmarker {
+public:
+  explicit StaticLookupBenchmarker(std::unique_ptr<HeaderMapImpl> impl)
+      : header_map_(std::move(impl)) {}
+  absl::optional<HeaderMapImpl::StaticLookupResponse> lookup(absl::string_view key) {
+    return header_map_->staticLookup(key);
+  }
+
+private:
+  std::unique_ptr<HeaderMapImpl> header_map_;
+};
+
+static void headerMapImplStaticLookups(benchmark::State& state,
+                                       std::unique_ptr<HeaderMapImpl> header_map,
+                                       const std::vector<std::string>& keys) {
+  int i = keys.size();
+  StaticLookupBenchmarker benchmarker{std::move(header_map)};
+  for (auto _ : state) {
+    UNREFERENCED_PARAMETER(_);
+    auto result = benchmarker.lookup(keys[--i]);
+    if (i == 0) {
+      i = keys.size();
+    }
+    benchmark::DoNotOptimize(result);
+  }
+}
+
+static std::vector<std::string> makeMismatchedHeaders() {
```

Comment:
Optional: you can make this a `static constexpr std::array<absl::string_view>` if you want -- that will not be a fiasco. Not sure if that would meet your use-case.

Reply:
It's a bit fiasco-ey because either it has to be turned into a vector anyway before passing it to the other function, or the other function has to take a template range of string-likes or something in order to support the *other* caller having a generated vector of std::string. Also having to manually specify the size to std::array feels gross. It's not like there's a performance issue generating a vector here, so simpler is better IMO.